src/elastic.jl

```diff
@@ -125,3 +133,28 @@ function elastic_worker(cookie, addr="127.0.0.1", port=9009; stdout_to_master=tr
     stdout_to_master && redirect_stdout(c)
     start_worker(c, cookie)
 end
+
+
+function get_private_ip()
+    if Sys.islinux()
+        IPv4(strip(read(`hostname --ip-address`, String)))
+    else
+        error("addr=:auto is only supported on Linux")
+    end
+end
+
+function get_connect_cmd(em::ElasticManager; absolute_exename=true, same_project=true)
+    
+    ip = string(em.sockname[1])
+    port = convert(Int,em.sockname[2])
+    cookie = cluster_cookie()
+    exename = absolute_exename ? joinpath(Sys.BINDIR, Base.julia_exename()) : "julia"
+    project = same_project ? ("--project=$(Pkg.API.Context().env.project_file)",) : ()
```

Comment:
Not sure this one function is sufficient reason to take a dependency on Pkg. And how sure are we that that implementation is stable?

Reply:
Thanks for looking this over. I'm not sure I understand the issue with taking a dependency. Everyone's Julia installation will always include `Pkg` and load time is tiny. Re stability, I can say at least that `Pkg.API.Context().env.project_file` seems to be the way `Pkg` itself looks up the current package to e.g. print when you do `Pkg.status()`.

Reply:
The dependency is not so much about loading time, and more about maintenance burden. The reason I ask about stability is because, if it's not exported/part of the stable API, then updates are free to break it without consideration. Given that this package doesn't have great (or any really) tests yet, it may be particularly important that we don't add too much maintenance burden. 

That said, I'm not the one to approve/reject, just wanted to flag it.

Reply:
as of julia 1.4 there is also `Pkg.project().name`.   for what version will `Pkg.API.Context().env.project_file` work?

Reply:
```
$ for ver in 1.0 1.1 1.2 1.3 1.4; do d run --rm julia:$ver julia -e "using Pkg; @show Pkg.API.Context().env.project_file"; done
(((Pkg.API).Context()).env).project_file = "/root/.julia/environments/v1.0/Project.toml"
(((Pkg.API).Context()).env).project_file = "/root/.julia/environments/v1.1/Project.toml"
(Pkg.API.Context()).env.project_file = "/root/.julia/environments/v1.2/Project.toml"
(Pkg.API.Context()).env.project_file = "/root/.julia/environments/v1.3/Project.toml"
(Pkg.API.Context()).env.project_file = "/root/.julia/environments/v1.4/Project.toml"
```
Everything back to the lowest supported version by this package (1.0) looks good, as does nightly which I just checked by hand.